File: rangeseekbar-library/src/main/java/it/sephiroth/android/library/rangeseekbar/DrawableUtils.java
package it.sephiroth.android.library.rangeseekbar;

import android.graphics.PorterDuff;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.DrawableContainer;
import android.graphics.drawable.DrawableWrapper;
import android.graphics.drawable.ScaleDrawable;
import android.os.Build;

import java.util.Objects;

import androidx.annotation.NonNull;

public class DrawableUtils {
    private static final String VECTOR_DRAWABLE_CLAZZ_NAME
        = "android.graphics.drawable.VectorDrawable";

    private DrawableUtils() { }

    public static PorterDuff.Mode parseTintMode(int value, PorterDuff.Mode defaultMode) {
        switch (value) {
            case 3:
                return PorterDuff.Mode.SRC_OVER;
            case 5:
                return PorterDuff.Mode.SRC_IN;
            case 9:
                return PorterDuff.Mode.SRC_ATOP;
            case 14:
                return PorterDuff.Mode.MULTIPLY;
            case 15:
                return PorterDuff.Mode.SCREEN;
            case 16:
                return PorterDuff.Mode.ADD;
            default:
                return defaultMode;
        }
    }

    static void fixDrawable(@NonNull final Drawable drawable) {
        if (Build.VERSION.SDK_INT == 21
            && VECTOR_DRAWABLE_CLAZZ_NAME.equals(drawable.getClass().getName())) {
            fixVectorDrawableTinting(drawable);
        }
    }

    private static void fixVectorDrawableTinting(final Drawable drawable) {
        final int[] originalState = drawable.getState();
        if (originalState == null || originalState.length == 0) {
            drawable.setState(ThemeUtils.CHECKED_STATE_SET);
        } else {
            drawable.setState(ThemeUtils.EMPTY_STATE_SET);
        }
        drawable.setState(originalState);
    }
}

File: rangeseekbar-library/src/main/java/it/sephiroth/android/library/rangeseekbar/DrawableUtils.kt
package it.sephiroth.android.library.rangeseekbar

import android.graphics.PorterDuff

object DrawableUtils {
    private const val VECTOR_DRAWABLE_CLAZZ_NAME = "android.graphics.drawable.VectorDrawable"

    fun parseTintMode(value: Int, defaultMode: PorterDuff.Mode): PorterDuff.Mode {
        when (value) {
            3 -> return PorterDuff.Mode.SRC_OVER
            5 -> return PorterDuff.Mode.SRC_IN
            9 -> return PorterDuff.Mode.SRC_ATOP
            14 -> return PorterDuff.Mode.MULTIPLY
            15 -> return PorterDuff.Mode.SCREEN
            16 -> return PorterDuff.Mode.ADD
            else -> return defaultMode
        }
    }
}
